app/controllers/api/v1/user_events_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Api::V1::UserEventsController < ApplicationController
+    def index
+        render json: UserEvent.all
+    end
+
+    def show
+        render json: UserEvent.find(params[:id])
+    end
+
+    def create
+        info = params[:event]
+        
+        if info.keys.include?('event_type') && info.keys.include?('name')
+
+            user_event = UserEvent.new(event_type: info['event_type'], event_data: remove_event_type(params))
+            user_event.save
+            render json: {status: "success", code: 200, message: "User Event Created"}
+        elsif 
```

Comment:
These next `elsif`s pertain to cases where both or one or the other of the necessary keys is missing. I would have liked to use the `errors.full_messages` but since the `name` was not a specific attribute of the `user_event` I just did it long-hand to get it to work as instructed.